wagtailio/project_styleguide/pattern_contexts.py

```diff
@@ -0,0 +1,9 @@
+from pattern_library import register_context_modifier
+
+from wagtailio.navigation.models import NavigationSettings
+
+
+@register_context_modifier
+def add_navigation(context, request):
+    if nav := NavigationSettings.objects.first():
```

Comment:
Note: this is not a full review. I see what you're trying to do here in terms of reducing queries but this will result in an error if a NavigationSettings instance is not found

Reply:
By using `.first()`, we avoid the error, since, `None` will  be returned if there's no instance.

Reply:
Do we want to be pulling in the real data for use in the pattern library? It seems like we'd be better off hard-coding something representative here and not touching the DB at all. This isn't a show-stopper for now, though.